### DISCUSSION TITLE
WebGPU is not working

### DISCUSSION THREAD

**Discussion by n303ef** [cefsharp/CefSharp#4662] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

120.1.80

### Operating System

Windows 11

### Architecture

x64

### .Net Version

4.7.2

### Implementation

WinForms

### Reproduction Steps

Open URL https://browserleaks.com/webgpu in CefSharp browser (any settings).

### Expected behavior

Should confirm WebGPU support and get available WebGPU info.

### Actual behavior

Website says:

No WebGPU adapters found

### Regression?

It worked CefSharp 117, but stopped working since 119 (didn't check in 118).

### Does this problem also occur in the CEF Sample Application

No

### Other information

Cefclient - no problem, WebGPU works.

CefSharp Browser is created with `WebGl = CefState.Enabled` (WebGL works).

**Comment by amaitland** [comment]
> Does the problem reproduce using the [MinimalExample](https://github.com/cefsharp/CefSharp.MinimalExample)?

Did you test with the https://github.com/cefsharp/CefSharp.MinimalExample as per the bug report template?

Testing with the minimal example works as expected for me, nobody else has reported an issue.

Does your app have an app.manifest as per https://github.com/cefsharp/CefSharp/blob/cefsharp/120/NuGet/Readme.txt#L8

**Reply by n303ef** [reply]
MinimalExample also doesn't work:
![screenshot](https://github.com/cefsharp/CefSharp/assets/40265129/eef332b6-7d0b-43de-8cef-74e07a2e24bf)

**Reply by n303ef** [reply]
Also checked on several other computers - Cefclient still ok, CefSharp.MinimalExample still no supported adapters.

**Reply by amaitland** [reply]
Actually I can reproduce this with the `MinimalExample`. It also reproduces with the latest `cefclient`.

https://cef-builds.spotifycdn.com/cef_binary_120.1.10%2Bg3ce3184%2Bchromium-120.0.6099.129_windows64_client.tar.bz2

Please confirm which version of `cefclient` you are using. You can load `chrome://version` in the browser if you are unsure.

**Comment by n303ef** [comment]
I've found solution. Two files are missing: 

dxil.dll
dxcompiler.dll

These files were created when I've built CEF from source, and they are not provided with CEF redist which is used by CefSharp NuGet package.
With these two files WebGPU works fine.

**Reply by amaitland** [reply]
`CEF` doesn't currently package those files. Please create an issue on https://github.com/chromiumembedded/cef/issues and see what the CEF maintainer says about including them in the CEF binary distribution.

**Reply by amaitland** [reply]
https://github.com/chromiumembedded/cef/issues/3646
